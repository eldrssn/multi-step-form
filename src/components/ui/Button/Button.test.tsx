import { fireEvent, render } from '@testing-library/react';
import { Button } from './';

describe('Actions of Button', () => {
  test('if action is next', () => {
    const { getByText, container } = render(<Button action="next" />);
    const button = getByText(/next step/i);
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('button_next');
    expect(container).toMatchSnapshot();
  });

  test('if action is back', () => {
    const { getByText } = render(<Button action="back" />);
    const button = getByText(/go back/i);
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('button_back');
  });

  test('if action is confirm', () => {
    const { getByText } = render(<Button action="confirm" />);
    const button = getByText(/confirm/i);
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('button_confirm');
  });
});

describe('Types of Button', () => {
  test('has no type in props', () => {
    const { getByText } = render(<Button action="next" />);
    const button = getByText(/next step/i);
    expect(button).toHaveAttribute('type', 'button');
  });

  test('has submit type on props', () => {
    const { getByText } = render(<Button action="next" type="submit" />);
    const button = getByText(/next step/i);
    expect(button).toHaveAttribute('type', 'submit');
  });
});

test('if Button is clicked', () => {
  const onClickMocked = jest.fn();
  const { getByText } = render(
    <Button action="next" onClick={onClickMocked} />
  );
  const button = getByText(/next step/i);

  fireEvent.click(button);
  expect(onClickMocked).toHaveBeenCalled();
});
