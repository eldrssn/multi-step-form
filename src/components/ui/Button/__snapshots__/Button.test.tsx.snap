// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions of Button if action is next 1`] = `
<div>
  <button
    class="button button_next"
    type="button"
  >
    Next Step
  </button>
</div>
`;
